pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -2170,7 +2219,7 @@ func (tmpl *Template) GetType() TemplateType {
 // IsPodType returns whether or not the template is a pod type
 func (tmpl *Template) IsPodType() bool {
 	switch tmpl.GetType() {
-	case TemplateTypeContainer, TemplateTypeScript, TemplateTypeResource:
+	case TemplateTypeContainer, TemplateTypeScript, TemplateTypeResource, TemplateTypeData:
```

Comment:
this is one of the problems of introducing new templates - you have to find every line in the codebase like this and add your new type